auth0/src/main/java/com/auth0/android/callback/BaseCallback.kt

```diff
@@ -26,14 +26,20 @@ package com.auth0.android.callback
 import com.auth0.android.Auth0Exception
 
 /**
- * Callback that receives a single value on success.
+ * Interface for all callbacks used with Auth0 API clients
  */
-//TODO [SDK-2185]: Merge this interface into Callback
-public interface BaseCallback<T, U : Auth0Exception> : Callback<U> {
+public interface BaseCallback<T, U : Auth0Exception> {
```

Comment:
Let's rename `BaseCallback` into `Callback`; and create a new no-op interface named `BaseCallback` that extends `Callback`. That way, while we can guide devs into using the preferred "Callback" class name, we won't introduce breaking changes.

Also, let's `@Deprecated` the introduced `BaseCallback` interface so those still using it can receive the _warning_ to migrate to `Callback`.